ligninkmc/plot_bond_formation.py

```diff
@@ -0,0 +1,430 @@
+                 markerfacecolor=COLORS[4], markeredgecolor=COLORS[4], label=C5O4, capsize=3, ecolor=COLORS[4])
+    plt.errorbar(x_axis, avg_bond_info[AO4], yerr=std_bond_info[AO4], linestyle='none', marker='.', markersize=10,
+                 markerfacecolor=COLORS[5], markeredgecolor=COLORS[5], label=AO4, capsize=3, ecolor=COLORS[5])
+    plt.errorbar(x_axis, avg_bond_info[C5C5], yerr=std_bond_info[C5C5], linestyle='none', marker='.', markersize=10,
+                 markerfacecolor=COLORS[6], markeredgecolor=COLORS[6], label=C5C5, capsize=3, ecolor=COLORS[6])
+    if len(x_axis) > 1:
+        plt.xscale('log')
+
+    [plt.gca().spines[i].set_linewidth(1.5) for i in ['top', 'right', 'bottom', 'left']]
+    plt.gca().tick_params(axis='both', which='major', labelsize=14, direction='in', pad=8, top=True, right=True,
+                          width=1.5, length=6)
+    plt.gca().tick_params(axis='both', which='minor', labelsize=14, direction='in', pad=8, top=True, right=True,
+                          width=1, length=4)
+    plt.ylabel('Bond Type Yield (%)', fontsize=14)
+    plt.xlabel('SG Ratio', fontsize=14)
+    plt.ylim([0.0, 1.0])
+    plt.legend(fontsize=14, loc='upper center', bbox_to_anchor=(1.2, 1.05), frameon=False)
+    plt.title(plot_title)
+    plt.savefig(plot_fname, bbox_inches='tight', transparent=True)
+    print(f"Wrote file: {plot_fname}")
+
+
+def get_avg_percent_bonds(bond_list, num_opts, adj_lists, num_trials):
+    """
+    Given adj_list for a set of options, with repeats for each option, find the avg and std dev of percent of each
+    bond type
+    :param bond_list: list of strings representing each bond type
+    :param num_opts: number of options specified (should be length of adj_lists)
+    :param adj_lists: list of lists of adjs: outer is for each option, inner is for each repeat
+    :param num_trials: number of repeats (should be length of inner adj_lists list)
+    :return: avg_bonds, std_bonds: list of floats, list of floats: for each option tested, the average and std dev
+                  of bond distributions (percentages)
+    """
+    analysis = []
+    for i in range(num_opts):
+        cur_adjs = adj_lists[i]
+        analysis.append([analyze_adj_matrix(cur_adjs[j]) for j in range(num_trials)])
+
+    bond_percents = {}
+    avg_bonds = {}
+    std_bonds = {}
+
+    for bond_type in bond_list:
+        bond_percents[bond_type] = [[analysis[j][i][BONDS][bond_type]/sum(analysis[j][i][BONDS].values())
+                                     for i in range(num_trials)] for j in range(num_opts)]
+        avg_bonds[bond_type] = [np.mean(bond_pcts) for bond_pcts in bond_percents[bond_type]]
+        std_bonds[bond_type] = [np.sqrt(np.var(bond_pcts)) for bond_pcts in bond_percents[bond_type]]
+    return avg_bonds, std_bonds
+
+
+def parse_cmdline(argv=None):
+    """
+    Returns the parsed argument list and return code.
+    :param argv: A list of arguments, or `None` for ``sys.argv[1:]``.
+    """
```

Comment:
Combining these can be a new issue to address in another pull. As I mentioned, I made the plotting one this weekend just for debugging purposes. If we want to keep it, I can combine. Otherwise, I can kill it.